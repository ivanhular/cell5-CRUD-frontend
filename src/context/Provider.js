import React, { createContext, useReducer } from 'react'
import { hobbyReducer } from './reducers/hobbyReducer'
import { hobbyInitialState } from './initialStates/hobbyInitialState'

export const GlobalContext = createContext({})

export const GlobalProvider = ({ children }) => {
  const [data, dispatch] = useReducer(hobbyReducer, hobbyInitialState)
  return (
    <GlobalContext.Provider value={{ data, dispatch }}>
      {children}
    </GlobalContext.Provider>
  )
}
