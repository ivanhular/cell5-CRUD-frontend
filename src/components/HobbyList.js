import React, { useContext, useEffect } from 'react'
import { Table } from 'react-bootstrap'
import HobbyListItem from './HobbyListItem'
import { GlobalContext } from '../context/Provider'
import { listHobby } from '../context/actions/hobby'
import { HOBBY_SORT_BY } from '../constants/actionTypes'

const HobbyList = () => {
  const {
    data: { hobbies },
    dispatch,
  } = useContext(GlobalContext)
  useEffect(() => {
    listHobby()(dispatch)
  }, [dispatch])

  const sortByHeader = (e) => {
    console.log(e.target.innerHTML.toLowerCase())
    dispatch({ type: HOBBY_SORT_BY, payload: e.target.innerHTML.toLowerCase() })
  }
  return (
    <>
      <Table striped bordered hover size='sm'>
        <thead>
          <tr>
            <th style={{ cursor: 'pointer' }} onClick={sortByHeader}>
              Title
            </th>
            <th style={{ cursor: 'pointer' }} onClick={sortByHeader}>
              Description
            </th>
            <th style={{ width: '15%' }}>Actions</th>
          </tr>
        </thead>
        <tbody>
          {hobbies.map((hobby) => (
            <HobbyListItem hobby={hobby} key={hobby._id} />
          ))}
        </tbody>
      </Table>
    </>
  )
}

export default HobbyList
