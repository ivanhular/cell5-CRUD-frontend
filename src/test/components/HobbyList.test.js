import React from 'react'
import { shallow, mount, render } from '../../enzyme'
import HobbyList from '../../components/HobbyList'
import { GlobalContext, GlobalProvider } from '../../context/Provider'
import { act } from 'react-dom/test-utils' // ES6

let wrapper

beforeEach(() => {
  const hobbyInitialState = {
    loading: false,
    success: false,
    deleteLoading: false,
    deleteSuccess: false,
    error: '',
    hobbies: [
      { _id: '1', title: 'basketball', description: 'test' },
      { _id: '2', title: 'tennis', description: 'test' },
      { _id: '3', title: 'golf', description: 'test' },
    ],
    modal: false,
    deleteModal: false,
    contentModal: false,
    toast: false,
    toastMessage: '',
    selected: '',
    queryWikiLoading: false,
    queryWikiSuccess: false,
    queryContent: '',
    queryError: '',
  }

  wrapper = render(
    <GlobalContext.Provider value={{ data: hobbyInitialState }}>
      <HobbyList />
    </GlobalContext.Provider>
  )
})

afterEach(() => {})

test('should render HobbyList', async () => {
  expect(wrapper).toMatchSnapshot()
  expect(wrapper.find('tbody tr')).toHaveLength(3)
})
