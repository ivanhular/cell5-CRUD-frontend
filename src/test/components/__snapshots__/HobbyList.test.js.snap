// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render HobbyList 1`] = `
<table
  class="table table-sm table-striped table-bordered table-hover"
>
  <thead>
    <tr>
      <th
        style="cursor:pointer"
      >
        Title
      </th>
      <th
        style="cursor:pointer"
      >
        Description
      </th>
      <th
        style="width:15%"
      >
        Actions
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        basketball
      </td>
      <td>
        test
      </td>
      <td>
        <button
          class="btn btn-success btn-sm"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-eye fa-w-18 "
            data-icon="eye"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M572.52 241.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400a144 144 0 1 1 144-144 143.93 143.93 0 0 1-144 144zm0-240a95.31 95.31 0 0 0-25.31 3.79 47.85 47.85 0 0 1-66.9 66.9A95.78 95.78 0 1 0 288 160z"
              fill="currentColor"
            />
          </svg>
        </button>
         
        <button
          class="btn btn-primary btn-sm"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-edit fa-w-18 "
            data-icon="edit"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
              fill="currentColor"
            />
          </svg>
        </button>
         
        <button
          class="btn btn-danger btn-sm"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-trash fa-w-14 "
            data-icon="trash"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
              fill="currentColor"
            />
          </svg>
        </button>
      </td>
    </tr>
    <tr>
      <td>
        tennis
      </td>
      <td>
        test
      </td>
      <td>
        <button
          class="btn btn-success btn-sm"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-eye fa-w-18 "
            data-icon="eye"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M572.52 241.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400a144 144 0 1 1 144-144 143.93 143.93 0 0 1-144 144zm0-240a95.31 95.31 0 0 0-25.31 3.79 47.85 47.85 0 0 1-66.9 66.9A95.78 95.78 0 1 0 288 160z"
              fill="currentColor"
            />
          </svg>
        </button>
         
        <button
          class="btn btn-primary btn-sm"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-edit fa-w-18 "
            data-icon="edit"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
              fill="currentColor"
            />
          </svg>
        </button>
         
        <button
          class="btn btn-danger btn-sm"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-trash fa-w-14 "
            data-icon="trash"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
              fill="currentColor"
            />
          </svg>
        </button>
      </td>
    </tr>
    <tr>
      <td>
        golf
      </td>
      <td>
        test
      </td>
      <td>
        <button
          class="btn btn-success btn-sm"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-eye fa-w-18 "
            data-icon="eye"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M572.52 241.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400a144 144 0 1 1 144-144 143.93 143.93 0 0 1-144 144zm0-240a95.31 95.31 0 0 0-25.31 3.79 47.85 47.85 0 0 1-66.9 66.9A95.78 95.78 0 1 0 288 160z"
              fill="currentColor"
            />
          </svg>
        </button>
         
        <button
          class="btn btn-primary btn-sm"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-edit fa-w-18 "
            data-icon="edit"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
              fill="currentColor"
            />
          </svg>
        </button>
         
        <button
          class="btn btn-danger btn-sm"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-trash fa-w-14 "
            data-icon="trash"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
              fill="currentColor"
            />
          </svg>
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;
